### PR TITLE
Add more GBS samples from graphs

### PR DESCRIPTION
**Pull request by trbromley**
This PR adds more of the datasets of GBS samples from graphs. It also incorporates the changes into the `data` module and polishes the module docstring.

### CODE REVIEW

File: strawberryfields/gbs/data.py [hunk 1]
```diff
@@ -198,20 +214,199 @@ def threshold(self) -> bool:
...
     _data_filename = "planted"
-    n_mean = 6
-    n_max = 14
+    n_mean = 8
+    n_max = 20
+    threshold = True
+
+
+class TaceAs(Dataset):
+    """Binding interaction graph for the TACE-AS complex :cite:`banchi2019molecular`.
```
[the remark this answers is not in the document]
Comment (ixfoduap) [reply strawberryfields/gbs/data.py]: We should also specify which nodes form the maximum clique.

File: strawberryfields/gbs/data.py [hunk 2]
```diff
@@ -198,20 +214,199 @@ def threshold(self) -> bool:
...
+
+
+class PHat(Dataset):
+    """Random graph created using the p-hat generator of :cite:`gendreau1993solving`.
+
+    This graph is the ``p_hat300-1`` graph of the `DIMACS
+    <http://iridia.ulb.ac.be/~fmascia/maximum_clique/DIMACS-benchmark>`__ dataset, which is a
+    collection of large graphs with cliques that are hard to find. The best known clique of
+    this 300-node graph is of size 8 and is composed of nodes: ``[53, 123, 180, 218, 246, 267, 270,
+    286]``. This graph is not visualized due to its size.
```
Comment (ixfoduap) [inline strawberryfields/gbs/data.py]:
```suggestion
    286]``. This graph is not visualized due to its large size.
```

File: strawberryfields/gbs/data.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (trbromley) [reply strawberryfields/gbs/data.py]: Nice! Have fixed

[the remark this answers is not in the document]
Comment (trbromley) [reply strawberryfields/gbs/data.py]: For the second comment, I think we can point the user to the tutorial, which focuses on this graph.

**Pull request merged**